jetpack.php

```diff
@@ -56,6 +56,37 @@ function jetpack_require_lib_dir() {
 	return JETPACK__PLUGIN_DIR . '_inc/lib';
 }
 
+/**
+ * THIS CODE WOULD NEED TO BE DUPLICATED IN EACH PLUGIN...
+ */
+if ( ! function_exists( 'jetpack_enqueue_library' ) ) {
+	global $jetpack_libraries;
+	if ( ! is_array( $jetpack_libraries ) ) {
+		$jetpack_libraries = array();
+	}
+	function jetpack_enqueue_library( $class_name, $version, $path ) {
+		global $jetpack_libraries;
+		if ( ! isset( $jetpack_libraries[ $class_name ] )
+		     || version_compare( $jetpack_libraries[ $class_name ] ['version'], $version, '<' )
+		) {
+			$jetpack_libraries[ $class_name ] = array( 'version' => $version, 'path' => $path );
+		}
+	}
+
+	// add the autoloader
+	spl_autoload_register( function ( $class_name ) {
+		global $jetpack_libraries;
+		if ( isset( $jetpack_libraries[ $class_name ] ) ) {
+			if ( ! did_action( 'plugins_loaded' ) ) {
```

Comment:
Why would we want to load the libraries after the plugins have loaded?

Reply:
That is the only way to know for sure that plugins are all loaded. 
If we don't wait till all the plugins load how can we be sure that we are loading the latest version?

Reply:
Thanks for elaborating, that makes sense to me.